src/menus/NavigationMenus.cpp

```diff
@@ -29,57 +29,45 @@ void NextOrPrevFrame(AudacityProject &project, bool forward)
    auto temp1 = AButton::TemporarilyAllowFocus();
    auto temp2 = ASlider::TemporarilyAllowFocus();
    auto temp3 = MeterPanel::TemporarilyAllowFocus();
-
-   auto &toolManager = ToolManager::Get( project );
-   auto botDock = toolManager.GetBotDock();
-
-
-   // Define the set of windows we rotate among.
-   static const unsigned rotationSize = 3u;
-
-   wxWindow *const begin [rotationSize] = {
+   
+   std::vector<wxWindow*> seq = {
       ProjectWindow::Get( project ).GetTopPanel(),
       &TrackPanel::Get( project ),
-      botDock,
-   };
-
-   const auto end = begin + rotationSize;
-
-   // helper functions
-   auto IndexOf = [&](wxWindow *pWindow) {
-      return std::find(begin, end, pWindow) - begin;
+      ProjectWindow::Get(project).GetEffectsWindow(),
+      ToolManager::Get( project ).GetBotDock(),
    };
 
    auto FindAncestor = [&]() {
       wxWindow *pWindow = wxWindow::FindFocus();
-      unsigned index = rotationSize;
-      while ( pWindow &&
-              (rotationSize == (index = IndexOf(pWindow) ) ) )
+      while (pWindow)
+      {
+         auto it = std::find(seq.cbegin(), seq.cend(), pWindow);
+         if(it != seq.cend())
+            return static_cast<size_t>(std::distance(seq.cbegin(), it));
          pWindow = pWindow->GetParent();
-      return index;
+      }
+      return seq.size();
    };
 
    const auto idx = FindAncestor();
-   if (idx == rotationSize)
+   if (idx == seq.size())
       return;
 
    auto idx2 = idx;
-   auto increment = (forward ? 1 : rotationSize - 1);
-
-   while( idx != (idx2 = (idx2 + increment) % rotationSize) ) {
-      wxWindow *toFocus = begin[idx2];
-      bool bIsAnEmptyDock=false;
-      if( idx2 != 1 )
-         bIsAnEmptyDock = ((idx2==0) ? toolManager.GetTopDock() : botDock)->
```

Comment:
See: https://bugzilla.audacityteam.org/show_bug.cgi?id=1564

I think this change may break that old bug fix, on Windows though not on Mac.  The first window in `seq` is not a `ToolDock` but the parent of one.  Drag all toolbars off the top dock so they float, and test it.

Reply:
See also #3419

Reply:
Thanks, will fix that

Reply:
@Paul-Licameli I tested it in current master (without my changes) as you said and both top and bottom panels are focusable, so even if I revert this check in my branch it wouldn't fix the issue.  Also tried after configuration reset, both docks aren't empty, though they are look empty

Reply:
I don't understand.  Top or bottom panel is meant to be skipped in the Ctrl-f6 cycle if there are no toolbars on the dock.

Are you saying that there was already a regression in this behavior, in master, for Windows?

If so, then don't try to fix the regression if your PR is not introducing it.  There should instead be an issue which I think should not be fixed for alpha.

Reply:
Now I see and understand your rewrite.